client/api/index.js

```diff
@@ -163,6 +163,44 @@ export default class WCStripeAPI {
 			} );
 	}
 
+	/**
+	 * Creates and confirms a setup intent.
+	 *
+	 * @param {string} paymentMethodId The id of the payment method.
+	 *
+	 * @return {Promise} Promise containing the setup intent.
+	 */
+	setupIntent( paymentMethodId ) {
+		return this.request(
+			this.getAjaxUrl( 'create_and_confirm_setup_intent' ),
+			{
+				action: 'create_and_confirm_setup_intent',
+				'wc-stripe-payment-method': paymentMethodId,
+				_ajax_nonce: this.options?.createAndConfirmSetupIntentNonce,
+			}
+		).then( ( response ) => {
+			if ( ! response.success ) {
+				throw response.data.error;
+			}
+
+			if ( response.data.status === 'succeeded' ) {
```

Comment:
> Is response.data.status the property we want to check here? Maybe we'd like to return the setup intent status with the response.

> I get the following 400 response below when this.getStripe().confirmCardSetup() is triggered with a regular successful 4242424242424242 card.

Both are great catches! Yes, the `response.data.status` is the correct property I intended to check here so I've pushed up [e2dd7c9](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2791/commits/e2dd7c93290afc161f5d715f456b5e40e1209fa7) to send the setup intent status which incidently fixes both of these issues.

The 400 bad request error was coming from trying to confirm an intent that has already been confirmed: ![image](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/2275145/a0db1be3-61ab-457e-a956-d88439cf85a4)

Previously what I was doing was creating and confirming the intent within PHP via AJAX call, then sending back the setup intent to JS which was then trying to confirm it again because [this condition wasn't passing](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/add/my-account-add-card-with-deferred-intent/client/api/index.js#L186-L201).

Now that I'm sending the status this now means we're properly handling the `requires_action` response from creating the setup intent which adds/fixes support for adding 3DS cards as well 😄